src/modules/geolocation/geolocation-directive.js

```diff
@@ -146,8 +146,8 @@ angular.module('anol.geolocation')
                         geolocation.setTracking(false);
                         const position = geolocation.getPosition();
                         const accuracyGeometry = geolocation.getAccuracyGeometry();
-                        const constrainedPosition = view.adjustCenter(position);
-                        if (position[0] !== constrainedPosition[0] || position[1] !== constrainedPosition[1]) {
+                        const extent = view.getProperties().extent;
+                        if (!containsCoordinate(extent, position)) {
```

Comment:
It looks like the import for the `containsCoordinate` method. You can take a look at the respective docs in openlayers (https://openlayers.org/en/latest/apidoc/module-ol_extent.html#.containsCoordinate) to see how this is done.